SYS_ATL/API.py

```diff
@@ -1070,6 +1070,19 @@ def double_fission(self, stmt_pat1, stmt_pat2, n_lifts=1):
 
         return Procedure(loopir, _provenance_eq_Procedure=self)
 
+    def remove_loop(self, loop_pattern):
```

Comment:
I think this name suggests an unsafe operation that outright _deletes_ some code. This is really a special case of loop unrolling where the upper bound doesn't matter (apart from being positive) and the body is checked for idempotence. I won't block the PR on this, but a major part of me wonders whether the user experience would be better to roll this into `unroll_loop`.

To see what I mean:

```
for i in seq(0, N): S  =(unroll)>  S0 S1 ... SN-1 (when N > 0)  =(drop idempotent)>  S0
```

Reply:
I see, I think that makes sense and I don't have a strong opinion with either interface. `remove_loop` might be slightly more explicit than `unroll` just magically removing the loop for the user? Not sure

Reply:
I don't see this as magic for `unroll`... there are three cases:

1. The user asked to unroll a loop of constant, positive, extent... do the obvious thing.
2. The user asked to unroll a loop of non-constant extent _and_ the loop body is idempotent... replace with `if extent > 0: s[i/extent]`
3. The user asked to unroll a loop that cannot be unrolled... `raise SchedulingError`.

Reply:
I see. Let's put this on the agenda on Monday? I'm fine with both, we can sort out `remove_loop` and merge it to `unroll` afterward if necessary.

Reply:
Sure, sounds good to me.